pkg/image/registry/imagestreamtag/rest.go

```diff
@@ -93,12 +135,54 @@ func (r *REST) Get(ctx kapi.Context, id string) (runtime.Object, error) {
 			Namespace:         kapi.NamespaceValue(ctx),
 			Name:              id,
 			CreationTimestamp: event.Created,
+			Annotations:       istagAnnotations,
+
+			// resource version is logically the later of image or imagestream
+			// with numbers, concatenating the two will alway produce a number larger than either one
+			// with strings, it will produce something unique.
+			ResourceVersion: stream.ResourceVersion + image.ResourceVersion,
 		},
 		Image: *imageWithMetadata,
 	}
 	return &ist, nil
 }
 
+func (r *REST) Update(ctx kapi.Context, obj runtime.Object) (runtime.Object, bool, error) {
+	istag, ok := obj.(*api.ImageStreamTag)
+	if !ok {
+		return nil, false, kapierrors.NewBadRequest(fmt.Sprintf("obj is not an ImageStreamTag: %#v", obj))
+	}
+
+	old, err := r.Get(ctx, istag.Name)
+	if err != nil {
+		return nil, false, err
+	}
+
+	if err := rest.BeforeUpdate(Strategy, ctx, obj, old); err != nil {
+		return nil, false, err
+	}
+
+	// we only allow updates of annotations, so lets find the correct image stream and update it.
+	name, tag, err := nameAndTag(istag.Name)
+	if err != nil {
+		return nil, false, err
+	}
+
+	imageStream, err := r.imageStreamRegistry.GetImageStream(ctx, name)
+	if imageStream.Spec.Tags == nil {
+		imageStream.Spec.Tags = map[string]api.TagReference{}
+	}
+	tagRef := imageStream.Spec.Tags[tag]
+	tagRef.Annotations = istag.Annotations
+	imageStream.Spec.Tags[tag] = tagRef
+	if _, err := r.imageStreamRegistry.UpdateImageStream(ctx, imageStream); err != nil {
+		return nil, false, err
+	}
+
+	newIST, err := r.Get(ctx, istag.Name)
```

Comment:
Refetching again is bad - you could get a different result.  Should extract out the bit that joins the two.